test/base/test_collector.py

```diff
@@ -210,16 +236,19 @@ def test_collector(gym_reset_kwargs) -> None:
     with pytest.raises(TypeError):
         c2.collect()
 
+    def create_env(i: int, t: str) -> Callable[[], NXEnv]:
+        return lambda: NXEnv(i, t)
+
     # test NXEnv
     for obs_type in ["array", "object"]:
-        envs = SubprocVectorEnv([lambda i=x, t=obs_type: NXEnv(i, t) for x in [5, 10, 15, 20]])
+        envs = SubprocVectorEnv([create_env(x, obs_type) for x in [5, 10, 15, 20]])
```

Comment:
this is less safe than before because of [late binding](https://docs.python-guide.org/writing/gotchas/#late-binding-closures) in functions, a common python gotcha. Pls pass as kwargs instead to bind immediately.

Reply:
Done